packages/btc/src/transaction/build.ts

```diff
@@ -298,11 +313,13 @@ export class TxBuilder {
     }
   }
 
-  async calculateFee(addressType: AddressType): Promise<number> {
-    const feeRate = this.feeRate ?? this.defaultFeeRate;
-    if (!feeRate) {
+  async calculateFee(addressType: AddressType, feeRate?: number): Promise<number> {
+    if (!feeRate && !this.feeRate) {
       throw new TxBuildError(ErrorCodes.INVALID_FEE_RATE, `${ErrorMessages[ErrorCodes.INVALID_FEE_RATE]}: ${feeRate}`);
     }
+    if (!feeRate) {
+      feeRate = this.feeRate!;
```

Comment:
Changing function parameters directly is not good coding practice. You can use a new variable like this: 
```typescript
const txFeeRate = feeRate ?? this.feeRate!
```